go-controller/pkg/ovn/obj_retry.go

```diff
@@ -711,6 +797,103 @@ func (oc *Controller) updateResource(objectsToRetry *retryObjs, oldObj, newObj i
+		}
+		return nil
+	case factory.EgressIPNamespaceType:
+		oldNamespace := oldObj.(*kapi.Namespace)
+		newNamespace := newObj.(*kapi.Namespace)
+		if err := oc.reconcileEgressIPNamespace(oldNamespace, newNamespace); err != nil {
+			return fmt.Errorf("unable to update egress IP matching namespace: %s, err: %v", newNamespace.Name, err)
+		}
+		return nil
+	case factory.EgressIPPodType:
+		oldPod := oldObj.(*kapi.Pod)
+		newPod := newObj.(*kapi.Pod)
+		if util.PodCompleted(newPod) {
+			if err := oc.reconcileEgressIPPod(oldPod, nil); err != nil {
+				return fmt.Errorf("unable to delete egress IP matching completed pod: %s/%s, err: %v",
+					oldPod.Name, oldPod.Namespace, err)
+			}
+		} else {
+			if err := oc.reconcileEgressIPPod(oldPod, newPod); err != nil {
+				return fmt.Errorf("unable to update egress IP matching pod: %s/%s, err: %v", newPod.Name, newPod.Namespace, err)
+			}
+		}
+		return nil
+	case factory.EgressNodeType:
+		oldNode := oldObj.(*kapi.Node)
+		newNode := newObj.(*kapi.Node)
+		// Initialize the allocator on every update,
+		// ovnkube-node/cloud-network-config-controller will make sure to
+		// annotate the node with the egressIPConfig, but that might have
+		// happened after we processed the ADD for that object, hence keep
+		// retrying for all UPDATEs.
+		if err := oc.initEgressIPAllocator(newNode); err != nil {
+			klog.V(5).Infof("Egress node initialization error: %v", err)
+		}
+		nodeEgressLabel := util.GetNodeEgressLabel()
+		oldLabels := oldNode.GetLabels()
+		newLabels := newNode.GetLabels()
+		_, oldHadEgressLabel := oldLabels[nodeEgressLabel]
+		_, newHasEgressLabel := newLabels[nodeEgressLabel]
+		// If the node is not labelled for egress assignment, just return
+		// directly, we don't really need to set the ready / reachable
+		// status on this node if the user doesn't care about using it.
+		if !oldHadEgressLabel && !newHasEgressLabel {
+			return nil
+		}
+		if oldHadEgressLabel && !newHasEgressLabel {
+			klog.Infof("Node: %s has been un-labelled, deleting it from egress assignment", newNode.Name)
+			oc.setNodeEgressAssignable(oldNode.Name, false)
+			if err := oc.deleteEgressNode(oldNode); err != nil {
+				return err
+			}
+			return nil
+		}
+		isOldReady := oc.isEgressNodeReady(oldNode)
+		isNewReady := oc.isEgressNodeReady(newNode)
+		isNewReachable := oc.isEgressNodeReachable(newNode)
+		oc.setNodeEgressReady(newNode.Name, isNewReady)
+		oc.setNodeEgressReachable(newNode.Name, isNewReachable)
+		if !oldHadEgressLabel && newHasEgressLabel {
+			klog.Infof("Node: %s has been labelled, adding it for egress assignment", newNode.Name)
+			oc.setNodeEgressAssignable(newNode.Name, true)
+			if isNewReady && isNewReachable {
+				if err := oc.addEgressNode(newNode); err != nil {
+					return err
+				}
+			} else {
+				klog.Warningf("Node: %s has been labelled, but node is not ready and reachable, cannot use it for egress assignment", newNode.Name)
```

Comment:
@flavio-fernandes @kyrtapz can we safely continue here (as it was in the original code)? Or should we retry?

Reply:
this one eIP is good but the node itself has prob so retry eIP won't/can't help when node conditions is better we will redo this and add egressIP to the node

Reply:
nit: labeled